taichi/program/async_engine.cpp

```diff
@@ -363,7 +365,9 @@ bool AsyncEngine::optimize_listgen() {
     bool keep = true;
     if (offload->task_type == OffloadedStmt::TaskType::listgen) {
       // keep
-    } else if (offload->task_type == OffloadedStmt::TaskType::clear_list) {
+    } else if (is_clear_list_task(offload)) {
+      // TODO: this only handles the case where the serial task contains exactly
+      // one ClearListStmt. Shall we also handle fused cases?
```

Comment:
Yes, we should handle fused cases, but given this function will be rewritten, no worries about that for now :-)